proposals/42-human-readable-mutable-names.md

```diff
@@ -0,0 +1,235 @@
+      - Node tries to resolve with the most specific resolver (if present for TLD), and then fall back to the global one.
+    - Visual Aid / config mock-up
+      ```json
+        "DNS": {
+          "Resolvers": {
+            "*.eth": "https://eth.link-or-cloudflare/path/to/dns-query",
+            "*.crypto": "https://unstoppablesomething.example.com/path/to/dns-query",
+            "*.libre": "https://www.opennic.org/path/to/their/dns-query", 
+            "*": "https://mozilla.cloudflare-dns.com/dns-query"
+          }
+        }
+      ```
+- DNSLink provides interop for human-readable names and enables independence from PKI and ICANN.
+    - Anyone can set up and run own naming system without our involvement, blessing or endorsement.
+    - No vendor-specific client is included. 
+- IPNS provides a reliable solution for cryptographically-verifiable mutable names that won't break, as long someone is re-publishing the latest record.
+    - IPNS re-publishing works seamlessly, out of the box, just like providing CIDs for blocks in local datastore (pinned or not).
+    - It is possible to pin IPNS address to keep it around and ensure it resolves even when original publisher is gone.
+        - Node follows IPNS record updates and keeps re-publishing latest one if original author disappears.
+        - Node follows content updates behind the IPNS record, and update content pin every time IPNS record changes. 
+            - It is possible to opt-out from this behavior to re-publish IPNS record for `en.wikipedia-on-ipfs.org` without pinning all the data.
+
+
+####  What does success look like?
+_Success means impact. How will we know we did the right thing?_
+
+<!--
+Provide success criteria. These might include particular metrics, desired changes in the types of bug reports being filed, desired changes in qualitative user feedback (measured via surveys, etc), etc.
+-->
+
+- We see % of `dnslink=/ipns/` going up.
+- We see more IPNS records on DHT
+- We no longer see requests for supporting new naming systems.
+
+#### Counterpoints &amp; pre-mortem
+_Why might this project be lower impact than expected? How could this project fail to complete, or fail to be successful?_
+ 
+- Configurable per TLDs [DNS over HTTPS](https://en.wikipedia.org/wiki/DNS_over_HTTPS) resolver for  may be not enough to mitigate concerns around centralized nature of ICANN's DNS.
+- IPNS may have other problems that block it from practical use in DNSLink records (eg. resolution still taking way longer than immutable path, making initial page load slow).
+
+#### Alternatives
+_How might this project’s intent be realized in other ways (other than this project proposal)? What other potential solutions can address the same need?_
+
+- Deprecate IPNS.
+- Announce ENS being the official solution for DNS names and replace every DNS lookup to go via ENS.
```

Comment:
Would customer actually be ok with this given the Gas fees to publish a new CID?

Reply:
Chicken and egg, of sorts: fixing IPNS would remove the problem of gas cost.

However, the way I see it,  the real problem here is not the cost of gas or chore of updating ENS record.
Using this as a global resolver would be just a charade: most domains will remain on ICANN network for years to come, so ENS  would act as _a centralized_ proxy for 99,9....% of ICANN lookups :see_no_evil: